From: Arnaud Rebillout <arnaudr@kali.org>
Date: Wed, 18 Oct 2023 16:30:49 +0700
Subject: monitor: Add missing error handling when marking a mirror down

---
 daemon/monitor.go | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/daemon/monitor.go b/daemon/monitor.go
index 282a6c6..dee5332 100644
--- a/daemon/monitor.go
+++ b/daemon/monitor.go
@@ -532,10 +532,14 @@ func (m *monitor) healthCheck(mirror mirrors.Mirror) error {
 		if opErr, ok := err.(*net.OpError); ok {
 			log.Debugf("Op: %s | Net: %s | Addr: %s | Err: %s | Temporary: %t", opErr.Op, opErr.Net, opErr.Addr, opErr.Error(), opErr.Temporary())
 		}
+		var err2 error
 		if strings.Contains(err.Error(), errRedirect.Error()) {
-			mirrors.MarkMirrorDown(m.redis, mirror.ID, "Unauthorized redirect")
+			err2 = mirrors.MarkMirrorDown(m.redis, mirror.ID, "Unauthorized redirect")
 		} else {
-			mirrors.MarkMirrorDown(m.redis, mirror.ID, "Unreachable")
+			err2 = mirrors.MarkMirrorDown(m.redis, mirror.ID, "Unreachable")
+		}
+		if err2 != nil {
+			log.Errorf(format+"Unable to mark mirror as down: %s", mirror.Name, err2)
 		}
 		log.Errorf(format+"Error: %s (%dms)", mirror.Name, err.Error(), elapsed/time.Millisecond)
 		return err
